main.go

```diff
@@ -21,63 +20,22 @@ func init() {
 	logger.Init()
 	log.Debug().Msgf("Logging level: %d", *config.EnvVars.LogLevel)
 
-	if err != nil {
-		log.Fatal().Err(err).Msg("failed to read env config")
-	}
-
 	var appsecret []byte
 	var dbsecret []byte
 
-	if config.EnvVars.Config == "" {
-		log.Debug().Msg("No config file specified, fetching secrets from vault")
-		vaultAddr := os.Getenv("VAULT_ADDR")
-		if vaultAddr == "" {
-			log.Fatal().Msg("VAULT_ADDR must be set")
-		}
-		vaultRole := os.Getenv("VAULT_ROLE")
-		if vaultRole == "" {
-			log.Fatal().Msg("VAULT_ROLE must be set")
-		}
-
-		appSecret := os.Getenv("APP_SECRET")
-		if appSecret == "" {
-			log.Fatal().Msg("APP_SECRET must be set")
-		}
-
-		dbSecret := os.Getenv("DB_SECRET")
-		if dbSecret == "" {
-			log.Fatal().Msg("DB_SECRET must be set")
-		}
-
-		// fetch app secrets
-		secretData, err := gcpss.FetchVaultSecret(vaultAddr, appSecret, vaultRole)
-		if err != nil {
-			log.Fatal().Err(err).Msgf("Unable to fetch secrets from vault. error %v", err)
-		}
-		appsecret = []byte(secretData)
-
-		// fetch db secrets
-		secretData, err = gcpss.FetchVaultSecret(vaultAddr, dbSecret, vaultRole)
-		if err != nil {
-			log.Fatal().Err(err).Msgf("Unable to fetch secrets from vault. error %v", err)
-		}
-		dbsecret = []byte(secretData)
-
-	} else {
-		log.Debug().Msgf("Using config file: %s", config.EnvVars.Config)
-		bytes, err := os.ReadFile(config.EnvVars.Config)
-		if err != nil {
-			log.Fatal().Err(err).Msgf("Unable to read file %s", config.EnvVars.Config)
-		}
-		appsecret = bytes
+	log.Debug().Msgf("Using config file: %s", config.EnvVars.Config)
+	bytes, err := os.ReadFile(config.EnvVars.Config)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Unable to read file %s", config.EnvVars.Config)
+	}
+	appsecret = bytes
 
-		log.Debug().Msgf("Using db config file: %s", config.EnvVars.DBConfig)
-		bytes, err = os.ReadFile(config.EnvVars.DBConfig)
-		if err != nil {
-			log.Fatal().Err(err).Msgf("Unable to read file %s", config.EnvVars.DBConfig)
-		}
-		dbsecret = bytes
```

Comment:
Just to be sure that I understand this part. Instead of using the Golang package to fetch the secrets, if not provided by flag, you will always look for a file?

Reply:
Yeah, moving that part to harpocrates means that we dont have to fetch secrets from the main container anymore. A bit more clean :)

Reply:
It's basically the same flow as when we deploy to k8s